backend/src/main/java/com/project/stocks/service/ScoreBuilder.java

```diff
@@ -45,27 +48,37 @@ public Score build() {
 
 
     public ScoreBuilder withOPM(List<YearInfo> opmList) {
-        score.addValue(calculateYearlyStatistics(opmList, Logic.Increasing));
+        int value = calculateYearlyStatistics(opmList, Logic.Increasing);
+        score.addValue(value);
+        score.getScoreBreakdown().setOperatingProfitMargin(value);
```

Comment:
How I would write this code would be to have a score breakdown property in the builder,.
Use `scoreBreakdown.setOperatingProfitMargin(value);` 

Then in the build method just add
`score.setScoreBreakdown(scoreBreakdown);`